lib/puppet/type/exec.rb

```diff
@@ -205,6 +209,12 @@ def sync
       end
     end
 
+    newparam(:pass_through_shell, :boolean => true, :parent => Puppet::Parameter::Boolean, :required_features => :supports_pass_through_shell) do
+      defaultto(:true)
+
+      newvalues(:true, :false)
+    end
+
```

Comment:
Could you add a `desc? Also maybe mention that we strongly recommend setting the default to false?